include/kernel.h

```diff
@@ -716,8 +716,10 @@ __syscall k_tid_t k_current_get(void);
  *
  * @retval 0 Thread spawning canceled.
  * @retval -EINVAL Thread has already started executing.
+ *
+ * @deprecated This API is deprecated.  Use k_thread_abort().
  */
-__syscall int k_thread_cancel(k_tid_t thread);
+ __deprecated __syscall int k_thread_cancel(k_tid_t thread);
```

Comment:
Shouldn't __syscall be the first thing in this line?

Reply:
I think this is actually the right way to do this. However, please check the generated syscall_dispatch.c to ensure it is there.

Reply:
Actually, it doesn't work if you put __syscall first.  The script appears to suck "__deprecated" into the function siganture of the implementation, which then spews warnings when referenced.  I think.  All I know is that I had it that way first, got warnings that killed the build, then swapped it to fix it. :)